src/CommandInvocationParser.cpp

```diff
@@ -18,6 +18,75 @@ namespace cmake::language
     {
       return { std::find_if_not(r.begin, r.end, [](char c) {return std::isspace(c); }), r.end };
     }
+
+    //-----------------------------------------------------------------------------
+    tl::expected<Token, Error> ParserIdentifier(Range r)
+    {
+      auto newRange = IgnoreSpaces(r);
+      auto identifier = Parser<ElementType::Identifier>{}.Parse(newRange);
+      if (identifier)
+      {
+        // Make sure we take into account the spaces
+        identifier->range.begin = r.begin;
+      }
+      return identifier;
+    }
+
+    //-----------------------------------------------------------------------------
+    tl::expected<Token, Error> ParserArguments(Range r)
+    {
+      auto newRange = IgnoreSpaces(r);
+      if (*newRange.begin != '(')
+      {
+        Error err;
+        err.message = "A command invocation should wrap its arguments between parenthesis.";
+        err.context = newRange.begin;
+        return tl::make_unexpected(err);
+      }
+
+      auto args = Parser<ElementType::Arguments>{}.Parse(Range{ newRange.begin + 1, r.end });
```

Comment:
General reflection maybe not directly for the PR here: I find it strange to remove the first parenthesis from what we pass to the arguments parser but to keep the end parenthesis.

Reply:
Here we do not know where the arguments ends, we only know `newRange.end+1` is the beginning of the arguments. `r.end` points to the end of the range, which usually points to the end of the script, not just the end of the command.